File: Hello_World_Tutorials/testing/c++/bst.cpp

#include <string>
#include "Bridges.h"
#include "DataSource.h"
#include "data_src/EarthquakeUSGS.h"
#include "BSTElement.h"


using namespace std;
using namespace bridges;

BSTElement<float, EarthquakeUSGS> *insert (BSTElement<float, EarthquakeUSGS> *rt,
	BSTElement<float, EarthquakeUSGS> *new_el);

int max_quakes = 25;

int main(int argc, char **argv) {
	Bridges *bridges =  new Bridges(110, "YOUR_USER_ID", "YOUR_API_KEY");


	string hilite_color = "orange",
		   def_color = "green",
		   end_color = "red";

	// read the earth quake  data and build the BST
	bridges->setTitle("Recent Earthquakes (USGIS Data)");

	DataSource *ds = new DataSource;

	vector<EarthquakeUSGS> eq_list = ds->getEarthquakeUSGSData(max_quakes);

	BSTElement<float, EarthquakeUSGS> *root = nullptr;

	for (int k = 0; k < max_quakes; k++) {
		EarthquakeUSGS eq = eq_list[k];
		BSTElement<float, EarthquakeUSGS>
		*bst_node = new BSTElement<float, EarthquakeUSGS>(eq.getMagnitude(), eq);
		bst_node->setLabel(eq.getTitle() + "\\nLat/Long: ( " +
			to_string(eq.getLatit()) + "," + to_string(eq.getLongit()) + " )\\n" +
			eq.getDateStr());
		root = insert (root, bst_node);
		root->getVisualizer()->setColor(Color("red"));
	}

	// visualize the binary search tree
	bridges->setDataStructure(root);
	bridges->visualize();

	return 0;
}

BSTElement<float, EarthquakeUSGS> *insert (BSTElement<float, EarthquakeUSGS> *rt,
	BSTElement<float, EarthquakeUSGS> *new_el) {
	if (rt == nullptr)
		return new_el;
	else if (new_el->getKey() < rt->getKey())
		rt->setLeft(insert(rt->getLeft(), new_el));
	else
		rt->setRight(insert(rt->getRight(), new_el));

	return rt;
}

File: Hello_World_Tutorials/testing/python/bst.py
from bridges.bridges import *
from bridges.bst_element import *
from bridges.data_src_dependent.data_source import *
import sys

# recursive insert method to insert nodes into a binary search tree
def insertR(rt, newel):
    if (rt is None):
        return newel
    elif newel.get_key() < rt.get_key():
        rt.set_left(insertR(rt.get_left(), newel))
    else:
        rt.set_right(insertR(rt.get_right(), newel))
    return rt


def main():
    args = sys.argv[1:]

    # create the Bridges object, set credentials
    bridges = Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

    bridges.set_title("A Binary Search Tree Example with Earthquake Data")

    # Retrieve a list of 10 earthquake records from USGS using the BRIDGES API
    ami = getEarthquakeUSGSData(10)

    root = None
    # create BST nodes and insert into a tree
    for i in range(len(ami)):
        bst_node = BSTElement(ami[i].getMagnitude(), ami[i])
        # set label of the node
        bst_node.set_label(ami[i].getTitle() + ami[i].getTime())

        root = insertR(root, bst_node)

    # set some visual attributes
    root.get_visualizer().set_color("red")

    #set visualizer type
    bridges.set_data_structure(root)
    #visualize the tree
    bridges.visualize()


if __name__ == "__main__":
    main()

